core/src/main/java/org/infinispan/statetransfer/StateTransferLock.java

```diff
@@ -39,15 +43,35 @@
    // transaction data latch
    void notifyTransactionDataReceived(int topologyId);
 
-   void waitForTransactionData(int expectedTopologyId, long timeout, TimeUnit unit) throws InterruptedException;
+   CompletableFuture<Void> transactionDataFuture(int expectedTopologyId);
+
+   @Deprecated
+   default void waitForTransactionData(int expectedTopologyId, long timeout, TimeUnit unit) throws InterruptedException, TimeoutException {
```

Comment:
Fair enough. I was thinking that you already had the methods in `StateTransferLockImpl`, but of course the code is quite different.